### PR TITLE
feat(adapters): add Spark on Gnosis

### PR DESCRIPTION
**Pull request by adamazad**
Adds Spark on Gnosis

Reference:

- [Ethereum addresses](https://devs.spark.fi/deployment-addresses/ethereum-addresses) 
- [Gnosis addresses](https://devs.spark.fi/deployment-addresses/gnosis-addresses)

### CODE REVIEW

**Comment by slasher125** [comment]
hi, ty for the update. did u test this locally? seems like there are few bugs

File: src/adaptors/spark/index.js [hunk 1]
```diff
@@ -6,50 +6,67 @@ const utils = require('../utils');
...
-    '0xFc21d6d146E6086B8359705C8b28512a983db0cb';
+const sparkChains = ['ethereum', 'gnosis'];
+
+/**
+ * protocol data provider address per chain
+ * @type {{ethereum: string, gnosis: string}}
+ */
+const protocolDataProviderAddress = {
+  ethereum: '0xFc21d6d146E6086B8359705C8b28512a983db0cb',
+  gnosis: '0xA98DaCB3fC964A6A0d2ce3B77294241585EAbA6d',
```
Comment (slasher125) [inline src/adaptors/spark/index.js]: not working with the imported abi, guess wrong contract

File: src/adaptors/spark/index.js [hunk 2]
```diff
@@ -79,74 +96,84 @@ const ethV3Pools = async () => {
...
   ).output;
-  const daiPool = ethPools.find(p=>p.symbol==="DAI")
-  daiPool.totalSupplyUsd = Number(ilk.line)/1e45
-  daiPool.tvlUsd = daiPool.totalSupplyUsd - daiPool.totalBorrowUsd
 
-  return ethPools;
+  const ethereumDaiPool = v3Pools.find(
+    (p) => p.symbol === 'DAI' && p.chain === 'Ethereum'
+  );
+  daiPool.totalSupplyUsd = Number(ilk.line) / 1e45;
```
Comment (slasher125) [inline src/adaptors/spark/index.js]: daiPool isn't declared

File: src/adaptors/spark/index.js (hunk 1 above)
Comment (adamazad) [reply src/adaptors/spark/index.js]: Yeah, the gnosis address was wrong, but still the test fails when reading from the eth's address

image

File: src/adaptors/spark/index.js (hunk 2 above)
Comment (adamazad) [reply src/adaptors/spark/index.js]: fixed in b34d3c0

File: src/adaptors/spark/index.js (hunk 1 above)
Comment (slasher125) [reply src/adaptors/spark/index.js]: its rpc related, is fine on server side!

**Comment by adamazad** [comment]
@slasher125, I don't see the yields on production yet.

**Pull request merged**